expyfun/_experiment_controller.py

```diff
@@ -280,9 +287,34 @@ def __init__(self, exp_name, audio_controller=None, response_device=None,
             # Initialize devices
             #
 
+            # Handle the tdt_delay parameter
+            if self._tdt_delay is not None:
+                if self.audio_type == 'tdt':
+                    try:
+                        np.isfinite(self._tdt_delay)  # bombs if not numeric
+                    except:
```

Comment:
never `except:` always `except Exception:` (the former even catches `SystemExit`). Also I think this might be good enough:
```
delay = int(tdt_delay)  # we can live with what this raises if it raises something
if delay < 0:
    raise ValueError(...)
```